pkg/build/api/types.go

```diff
@@ -397,6 +397,9 @@ const (
 	// availability of a new version of an image
 	ImageChangeBuildTriggerType           BuildTriggerType = "ImageChange"
 	ImageChangeBuildTriggerTypeDeprecated BuildTriggerType = "imageChange"
+
+	// ConfigChangeBuildTriggerType will trigger a build on an initial build config creation
+	ConfigChangeBuildTriggerType BuildTriggerType = "ConfigChange"
```

Comment:
I'm fine with documenting the current and eventual behavior, so we don't have to do wacky things in the future to keep the initial behavior